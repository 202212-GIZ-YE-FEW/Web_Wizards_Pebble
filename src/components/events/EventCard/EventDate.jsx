const EventDate = ({ eventDate, icon }) => {
    function formatDate(dateObj) {
        const date = new Date(
            dateObj?.seconds * 1000 + dateObj?.nanoseconds / 1000000
        );
        const options = {
            weekday: "short",
            month: "short",
            day: "numeric",
            hour: "numeric",
            minute: "numeric",
            timeZoneName: "short",
        };
        return date.toLocaleString("en-US", options);
    }
    const formattedDate = formatDate(eventDate);

    return (
        <div className='flex items-center'>
            {icon && <span className='mb-3 mr-2 text-black-100'>{icon}</span>}
            <span>{formattedDate}</span>
        </div>
    );
};

export default EventDate;
