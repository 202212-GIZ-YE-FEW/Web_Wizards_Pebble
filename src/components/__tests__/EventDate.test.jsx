import renderer from "react-test-renderer";

import EventDate from "@/components/events/EventCard/EventDate";

it("renders correctly", () => {
    const tree = renderer.create(<EventDate />).toJSON();
    expect(tree).toMatchSnapshot();
});
