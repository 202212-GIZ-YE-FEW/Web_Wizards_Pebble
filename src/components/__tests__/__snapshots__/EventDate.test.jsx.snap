// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex items-center"
>
  <span
    className="ml-2"
  >
    Invalid Date
  </span>
</div>
`;
